pkg/kubelet/kubelet_node_status.go

```diff
@@ -172,6 +175,38 @@ func (kl *Kubelet) reconcileHugePageResource(initialNode, existingNode *v1.Node)
 	return requiresUpdate
 }
 
+// reconcileSwapResource will update swap capacity and remove swap size no longer supported
+func (kl *Kubelet) reconcileSwapResource(initialNode, existingNode *v1.Node) bool {
+	requiresUpdate := updateDefaultResources(initialNode, existingNode)
+	initialCapacity := initialNode.Status.Capacity[v1.ResourceSwap]
+
+	initialAllocatable := initialNode.Status.Allocatable[v1.ResourceSwap]
+
+	capacity, resourceIsCapacity := existingNode.Status.Capacity[v1.ResourceSwap]
+	allocatable := existingNode.Status.Allocatable[v1.ResourceSwap]
+
+	if resourceIsCapacity && !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NodeSwap) {
```

Comment:
This seems late for a feature gate check... I'd expect this to be the first thing we'd look for and then bail if it's unset. (Deleting this from the existing node status when flag is off seems reasonable.)